lib/mongodb.js

```diff
@@ -734,8 +734,13 @@ MongoDB.prototype.buildSort = function(model, order) {
 };
 
 function buildNearFilter(query, near) {
+  var node = query.where;
+  near.key.forEach(function(key) {
+    node = node[key];
+  });
+
```

Comment:
Did you mean to have the below code in this loop? Otherwise, what's the point of looping through the keys?